### DISCUSSION TITLE
Scheduler

### DISCUSSION THREAD

**Discussion by JustSteveKing** [minicli/minicli#62] [Ideas]
I had an idea for adding scheduled commands to Minicli - but as it's a large feature to add in I thought it best to post here first!

It would mirror Laravels scheduled commands in effect.

**Comment by JustSteveKing** [comment]
@WendellAdriel hey!

**Reply by WendellAdriel** [reply]
Thanks, @JustSteveKing 
I didn't see the discussions 😅 
Should I post here the ideas I sent in the issue?

**Reply by JustSteveKing** [reply]
Yeah! Or you can just link it here for ease? I'm sure we can all click through to read more of need be

**Reply by WendellAdriel** [reply]
Done, thanks man!

**Comment by WendellAdriel** [comment]
I think that this feature would be AWESOME in Minicli.
I have some ideas on how this could be built.
It's much simpler and not powerful as the Laravel Scheduled Commands, but I think it could already be a starting point.
I described the ideas **[here](https://github.com/minicli/minicli/issues/59#issuecomment-1534573374)**.

**Comment by erikaheidi** [comment]
This is a great idea, it would be very helpful 👏🏼 I think we would need another entry point for the scheduler, and some nice API to do things like `->weekly()` or something similar, instead of the traditional cron strings...

**Reply by JustSteveKing** [reply]
Yeah, we definitely want to avoid cron strings! Just because it is dependency free doesn't mean the DX needs to suck, right!? I will coordinate with @WendellAdriel and get this started.

**Reply by WendellAdriel** [reply]
True 😅 
The cron string was a starting idea, but we sure can create something better for DX!
Great, @JustSteveKing. Let me know how can I help and I'll be happy to help
